components/ui/Button.tsx

```diff
@@ -1,76 +1,185 @@
-import { StyleSheet, TextStyle, TouchableOpacity, TouchableOpacityProps, ViewStyle } from "react-native";
+import {
+  StyleSheet,
+  TextStyle,
+  TouchableOpacity,
+  TouchableOpacityProps,
+  ViewStyle,
+} from "react-native";
 import CustomText from "@/components/ui/CustomText";
 import { ButtonVariant, SizeVariants } from "@/types";
 import { theme } from "@/theme";
 interface ButtonProps extends TouchableOpacityProps {
-    variant?: ButtonVariant;
-    textSize: SizeVariants;
-    title: string;
-    active?: boolean,
-    containerStyles?: ViewStyle,
-    textStyles?: TextStyle
+  variant?: ButtonVariant;
+  textSize?: SizeVariants;
+  title?: string;
+  active?: boolean;
+  containerStyles?: ViewStyle;
+  textStyles?: TextStyle;
+  disabled?: boolean;
+  fullWidth?: boolean;
+  children?: React.ReactNode;
 }
-export function CustomButton(
-    {
-        variant = 'primary',
-        onPress,
-        textSize,
-        textStyles = {},
-        containerStyles = {},
-        active,
-        title,
-        ...rest
-    }: ButtonProps): JSX.Element {
-
-    const variantStyles = {
-        primary: {
-            container: styles.primary,
-            text: styles.primaryVariantText,
+export function CustomButton({
+  variant = "default",
+  onPress,
+  textSize,
+  textStyles = {},
+  containerStyles = {},
+  active,
+  disabled,
+  children,
+  fullWidth,
+  title,
+  ...rest
+}: ButtonProps): JSX.Element {
+  const getVariantStyles = (
+    variant: string
+  ): { container: ViewStyle; text?: TextStyle } => {
+    if (variant === "primary") {
+      return {
+        container: styles.primary,
+        text: styles.primaryText,
+      };
+    }
+    if (variant === "secondary") {
+      return {
```

Comment:
why not keep the previous format for the gettting the variant styles, I think it was more readable than using the if conditions and would probably use less lines of code. something like this,
```suggestion

const variantStyles = {
    primary: {
      container: styles.primary,
      text: styles.primaryVariantText,
    },
    secondary: {
      container: {
        ...(fullWidth && styles.secondaryFullWidth),
        ...styles.secondary,
      },
      text: styles.secondaryText,
    },
    outline: { container: styles.outline, text: styles.outlineText },
    destructive: {
      container: styles.destructive,
      text: styles.descructiveText,
    },
    ghost: {
      container: styles.ghost,
      text: styles.destructive,
    },
  };
const { container, text } = variantStyles[variant] || variantStyles.primary;
```